be/src/exec/pipeline/olap_chunk_source.cpp

```diff
@@ -420,9 +420,9 @@ Status OlapChunkSource::_read_chunk_from_storage(RuntimeState* state, vectorized
     }
     SCOPED_TIMER(_scan_timer);
     do {
-        if (Status status = _prj_iter->get_next(chunk); !status.ok()) {
-            return status;
-        }
+        RETURN_IF_ERROR(state->check_mem_limit("read chunk from storage"));
```

Comment:
Checking per chunk will affect query performance?

Reply:
https://github.com/StarRocks/starrocks/blob/5d3c5fc07d3fb2cad3bf7cc57a252d0b08e26214/be/src/exec/vectorized/aggregate/aggregate_blocking_node.cpp#L40

The performance has been tested before, and it does not affect the performance. Currently, Aggr is also fetch a Chunk and Check after every GetNext. These are temporary solutions. When the storage layer code supports TryCatch, it can be removed.

Reply:
Let me test the impact of high concurrency for small query, and then decide whether to do it